packages/i18n/src/index.js

```diff
@@ -1,172 +1,7 @@
- *
- * @param {string} text     Text to translate.
- * @param {string} [domain] Domain to retrieve the translated text.
- *
- * @return {string} Translated text.
- */
-export function __( text, domain ) {
-	return dcnpgettext( domain, undefined, text );
-}
-
-/**
- * Retrieve translated string with gettext context.
- *
- * @see https://developer.wordpress.org/reference/functions/_x/
- *
- * @param {string} text     Text to translate.
- * @param {string} context  Context information for the translators.
- * @param {string} [domain] Domain to retrieve the translated text.
- *
- * @return {string} Translated context string without pipe.
- */
-export function _x( text, context, domain ) {
-	return dcnpgettext( domain, context, text );
-}
-
-/**
- * Translates and retrieves the singular or plural form based on the supplied
- * number.
- *
- * @see https://developer.wordpress.org/reference/functions/_n/
- *
- * @param {string} single   The text to be used if the number is singular.
- * @param {string} plural   The text to be used if the number is plural.
- * @param {number} number   The number to compare against to use either the
- *                          singular or plural form.
- * @param {string} [domain] Domain to retrieve the translated text.
- *
- * @return {string} The translated singular or plural form.
- */
-export function _n( single, plural, number, domain ) {
-	return dcnpgettext( domain, undefined, single, plural, number );
-}
-
-/**
- * Translates and retrieves the singular or plural form based on the supplied
- * number, with gettext context.
- *
- * @see https://developer.wordpress.org/reference/functions/_nx/
- *
- * @param {string} single   The text to be used if the number is singular.
- * @param {string} plural   The text to be used if the number is plural.
- * @param {number} number   The number to compare against to use either the
- *                          singular or plural form.
- * @param {string} context  Context information for the translators.
- * @param {string} [domain] Domain to retrieve the translated text.
- *
- * @return {string} The translated singular or plural form.
- */
-export function _nx( single, plural, number, context, domain ) {
-	return dcnpgettext( domain, context, single, plural, number );
-}
-
-/**
- * Returns a formatted string. If an error occurs in applying the format, the
- * original format string is returned.
- *
- * @param {string}    format The format of the string to generate.
- * @param {...string} args   Arguments to apply to the format.
- *
- * @see http://www.diveintojavascript.com/projects/javascript-sprintf
- *
- * @return {string} The formatted string.
- */
-export function sprintf( format, ...args ) {
-	try {
-		return sprintfjs.sprintf( format, ...args );
-	} catch ( error ) {
-		logErrorOnce( 'sprintf error: \n\n' + error.toString() );
-
-		return format;
-	}
-}
+export { sprintf } from './sprintf';
+export { I18n } from './i18n';
+export * from './default-i18n';
```

Comment:
I'd personally prefer the more explicit
```js
export { i18n, setLocaleData, __, _x, _n, _nx } from './default-i18n';
```
then, when I want an authoritative answer about what are the exports of the `i18n` module, I only need to look at one file.

Reply:
I personally prefer the `*` syntax because it doesn't create a bunch of identifiers that have to be kept in sync with another file and it means there's no chance of forgetting to export something.

I've changed the code to suit your preference.